File: ccc01s2.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    
    int n;
    cin >> n;

    for (int x = 0; x < n; x++) {
        int curr, end;
        cin >> curr >> end;

        vector<vector<int>> grid = {{curr}};

        int moves = 0;

        while (curr != end) {
            curr += 1;

            if (moves % 2 == 0) {
                if (moves % 4 == 0) {
                    for (int i = 0; i < int(grid.size());i++, curr++) grid.push_back({curr});
                }
                else {
                    for (int i = 0; i < int(grid.size());i++, curr++) grid.emplace(grid.begin(), curr);
                }
            }
            else {
                if ((moves-1) % 4 == 0) {
                    for (int i = 0; i < int(grid[0].size());i++, curr++) grid[grid.size()-1].push_back(curr);
                }
                else {
                    for (int i = 0; i < int(grid[0].size());i++, curr++) grid[0].emplace(grid[0].begin(), curr);
                }
            }
            moves += 1;
        }

        for (auto row : grid) {
            for (auto col : row) {
                cout << col << " ";
            }
            cout << endl;
        }
        cout << endl;
    }
}

File: ccc01s2.py
n = int(input())

for x in range(n):

    temp = [int(x) for x in input().split(" ")]
    curr = temp[0]
    end = temp[1]

    grid = [[curr]]
    moves = 0
    curr += 1
    while (curr <= end):

        if (moves % 2 == 0):
            if (moves % 4 == 0):
                temp = len(grid)
                for i in range(temp-1):
                    grid[i+1].insert(0, " ")
                for i in range(temp-1):
                    grid[i+1][0] = curr
                    curr += 1
                grid.append([curr])
                curr += 1
            else:
                temp = len(grid)
                for i in range(temp-1):
                    grid[temp-i-2].append(curr)
                    curr += 1
                grid.insert(0, [curr])
                curr += 1

        else:
            if (moves % 4 == 1):
                temp = len(grid[0])
                for i in range(temp):
                    grid[-1].append(curr)
                    curr += 1
            else:
                temp = len(grid[0])
                for i in range(temp+1):
                    grid[0].insert(0, curr)
                    curr += 1

        moves += 1

    for i in range(len(grid[1]) - len(grid[0])+1):
        grid[0].insert(0, " ")

    for row in grid:
        for col in row:
            if col != " " and col > end:
                row.remove(col)

    for row in grid:
        for col in row:

            print(col, end=" ")
        print("")
    print("")